stripped/colabs/agregacje_kilkuletnie_stripped.py

```diff
@@ -0,0 +1,669 @@
+y = monthly_avg_4_july2023_an_merge['pustynia']
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=2023)
+
+lgbm_monthly_avg_4_july2023 = lgb.LGBMClassifier(max_depth=10, n_estimators=120, num_leaves=30, random_state=3)
+
+lgbm_monthly_avg_4_july2023.fit(X_train, y_train)
+
+"""Zastosujmy model na pełnych danych:"""
+
+monthly_avg_4_july2023_pred_lgbm = monthly_avg_4_july2023.copy()
+monthly_avg_4_july2023_pred_lgbm['pustynia'] = lgbm_monthly_avg_4_july2023.predict(monthly_avg_4_july2023.loc[:, [col for col in monthly_avg_4_july2023.columns if col.startswith('monthly_avg')]])
+
+output_notebook()
+show(plot_map(df=monthly_avg_4_july2023_pred_lgbm, parameter_name='pustynia',
+              colormap=dict(zip(['1', '0'], ['yellow', 'green'])),
+              title='Pustynie (1) i niepustynie (0) - LGBM (lipiec 2023, średnia 4-letnia)',
+              point_size=3, alpha=0.7))
+
+"""##### Ocena na zbiorze treningowym"""
+
+summary_model(lgbm_monthly_avg_4_july2023, X_train, y_train, ['0','1'])
+
+print_classification_report(lgbm_monthly_avg_4_july2023, X_train, y_train)
+
+show_metrics(lgbm_monthly_avg_4_july2023, X_train, y_train)
+
+"""##### Ocena na zbiorze testowym"""
+
+summary_model(lgbm_monthly_avg_4_july2023, X_test, y_test, ['0', '1'])
+
+print_classification_report(lgbm_monthly_avg_4_july2023, X_test, y_test)
+
+show_metrics(lgbm_monthly_avg_4_july2023, X_test, y_test)
+
+"""### Średnie 5-letnie"""
+
+X = monthly_avg_5_july2023_an_merge.drop(columns=['lon', 'lat', 'pustynia'])
+y = monthly_avg_5_july2023_an_merge['pustynia']
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=2023)
+
+lgbm_monthly_avg_5_july2023 = lgb.LGBMClassifier(max_depth=10, n_estimators=120, num_leaves=30, random_state=3)
+
+lgbm_monthly_avg_5_july2023.fit(X_train, y_train)
+
+"""Zastosujmy model na pełnych danych:"""
+
+monthly_avg_5_july2023_pred_lgbm = monthly_avg_5_july2023.copy()
+monthly_avg_5_july2023_pred_lgbm['pustynia'] = lgbm_monthly_avg_5_july2023.predict(monthly_avg_5_july2023.loc[:, [col for col in monthly_avg_5_july2023.columns if col.startswith('monthly_avg')]])
+
+output_notebook()
+show(plot_map(df=monthly_avg_5_july2023_pred_lgbm, parameter_name='pustynia',
+              colormap=dict(zip(['1', '0'], ['yellow', 'green'])),
+              title='Pustynie (1) i niepustynie (0) - LGBM (lipiec 2023, średnia 5-letnia)',
+              point_size=3, alpha=0.7))
+
+"""##### Ocena na zbiorze treningowym"""
+
+summary_model(lgbm_monthly_avg_5_july2023, X_train, y_train, ['0','1'])
+
+print_classification_report(lgbm_monthly_avg_5_july2023, X_train, y_train)
+
+show_metrics(lgbm_monthly_avg_5_july2023, X_train, y_train)
+
+"""##### Ocena na zbiorze testowym"""
+
+summary_model(lgbm_monthly_avg_5_july2023, X_test, y_test, ['0', '1'])
+
+print_classification_report(lgbm_monthly_avg_5_july2023, X_test, y_test)
+
+show_metrics(lgbm_monthly_avg_5_july2023, X_test, y_test)
+
+"""## Wnioski
```

Comment:
Modele wyszły dużo lepsze niż te tworzone na pojedynczych miesiącach, widać to choćby po mapach. Do ewentualnych dalszych analiz można by je zapisać w katalogu na dysku Google.

Reply:
Już dodałam do skryptu zapisywanie najlepszego modelu i poszło też na nasz dysk